src/test/java/seedu/address/logic/commands/DeleteTagFromTaskCommandTest.java

```diff
@@ -0,0 +1,92 @@
+import seedu.address.model.AddressBook;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.StrategyBoard;
+import seedu.address.model.TaskBook;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.tag.Tag;
+import seedu.address.model.task.Task;
+import seedu.address.testutil.TaskBuilder;
+
+class DeleteTagFromTaskCommandTest {
+    // Test tags
+    // Because PersonBuilder#withTags takes String ... instead of Set<Tag>
+    private static final String TAG1 = "colleagues";
+
+    // Test model
+    private Model model = new ModelManager(getTypicalAddressBook(), getTypicalTaskBook(),
+            new StrategyBoard(), new UserPrefs());
+
+    @Test
+    void execute_deleteTagCommandUnfilteredList_success() {
+        Task firstTask = model.getFilteredTaskList().get(TASK_FIRST_INDEX.getZeroBased());
+
+        // Adding 1 more tag to the firstTask
+        Set<Tag> firstTaskTags = new HashSet<>(firstTask.getTags()); // Set<Tag> copy of Meetings (first task)
+        firstTaskTags.remove(new Tag(TAG1));
+
+        // Convert Set<Tag> to array for TaskBuilder#withTags
+        String[] firstTaskTagsStringArray = firstTaskTags
+                .stream()
+                .map(x -> x.tagName)
+                .toArray(String[]::new);
+
+        // Manually building the edited task
+        Task editedTask = new TaskBuilder(firstTask).withTags(firstTaskTagsStringArray).build();
+
+        DeleteTagFromTaskCommand deleteTagFromTaskCommand = new DeleteTagFromTaskCommand(TASK_FIRST_INDEX, TAG1);
+
+        String expectedMessage = String.format(deleteTagFromTaskCommand.MESSAGE_DEL_TAG_SUCCESS, TAG1);
+
+        // Manually building the expected model
+        Model expectedModel = new ModelManager(
+                new AddressBook(model.getAddressBook()),
+                new TaskBook(model.getTaskBook()),
+                new StrategyBoard(),
+                new UserPrefs());
+        expectedModel.setTask(firstTask, editedTask);
+
+        assertCommandSuccess(deleteTagFromTaskCommand, model, expectedMessage, expectedModel);
+
+    }
+
+    @Test
+    public void execute_invalidTagIndexFilteredList_failure() {
+        showTaskAtIndex(model, TASK_FIRST_INDEX);
+        Index outOfBoundIndex = TASK_SECOND_INDEX;
+        // ensures that outOfBoundIndex is still in bounds of address book list
+        assertTrue(outOfBoundIndex.getZeroBased() < model.getTaskBook().getTaskList().size());
+
+        AddTagToTaskCommand addTagToTaskCommand = new AddTagToTaskCommand(outOfBoundIndex, TAG1);
+
+        assertCommandFailure(addTagToTaskCommand, model, Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void equals() {
+        final AddTagToTaskCommand command = new AddTagToTaskCommand(INDEX_FIRST_PERSON, TAG1);
+
+        // If they are the same objects, they are equal
+        assertTrue(command.equals(command));
+
+    }
```

Comment:
Consider adding more assertions here for the `equals()` method to increase test coverage.

Specifically, consider creating an additional assertion to check that two commands of equal `index` and `tagName` return true when using `equals()` on them.

Reply:
Added